### PR TITLE
Surface empty BCPT results as a warning instead of silent success

### LINKED ISSUE #2158: [Bug]: Performance Tests does not return any output and build is successful
### AL-Go version

8.3

### Describe the issue

BCPT tests exits without any output, even the simplest ones.

2026-03-06T17:54:58.1706344Z Using testpage 149002
2026-03-06T17:54:58.1707059Z Using Suitecode 10UserTest
2026-03-06T17:54:58.1715234Z Service Url http://bc22774581371/BC/cs/?tenant=default&company=CRONUS%20USA%2C%20Inc.
2026-03-06T17:54:58.1716356Z Running Performance Tests for 5 minutes...
2026-03-06T17:57:36.5067503Z Invoke GET on http://172.19.110.106:7048/BC/api/Microsoft/PerformancToolkit/v1.0/companies(c2f9be77-e218-f111-ac6c-7ced8dae1517)/bcptLogEntries?tenant=default
2026-03-06T17:57:36.6559956Z Saving bcpt test results to D:\a\DemoApp2\DemoApp2\bcptTestResults.json
2026-03-06T17:57:36.6610273Z 
2026-03-06T17:57:36.6610801Z Running BCPT tests took 171 seconds

It should run for 5 mins but exit after less than 3 minutes and with no output to show

[logs_59655157876.zip](https://github.com/user-attachments/files/25801474/logs_59655157876.zip)

- 

### Expected behavior

When running above scenario, a bcpt run output is expected to be shown 

### Steps to reproduce

- Run Action :  create a new app (use the default settings)
- Run Action : create a new performance test app (use the default settings)
- Change  "bcptTestFolders": ["MyApp.PerformanceTest"] in settings

### Additional context (logs, screenshots, etc.)

Attached logs

This is the relevant part>
2026-03-06T17:54:58.1706344Z Using testpage 149002
2026-03-06T17:54:58.1707059Z Using Suitecode 10UserTest
2026-03-06T17:54:58.1715234Z Service Url http://bc22774581371/BC/cs/?tenant=default&company=CRONUS%20USA%2C%20Inc.
2026-03-06T17:54:58.1716356Z Running Performance Tests for 5 minutes...
2026-03-06T17:57:36.5067503Z Invoke GET on http://172.19.110.106:7048/BC/api/Microsoft/PerformancToolkit/v1.0/companies(c2f9be77-e218-f111-ac6c-7ced8dae1517)/bcptLogEntries?tenant=default
2026-03-06T17:57:36.6559956Z Saving bcpt test results to D:\a\DemoApp2\DemoApp2\bcptTestResults.json
2026-03-06T17:57:36.6610273Z 
2026-03-06T17:57:36.6610801Z Running BCPT tests took 171 seconds

It should run for 5 mins but exit after less than 3 minutes and with no output to show

### PR DESCRIPTION
**Pull request by jeffreybulanadi**
## Problem

When BCPT performance tests run but produce no log entries (for example, because the test runner exits early, the suite codeunit IDs do not match the published app, or the job fails silently), the results file contains no entries. \GetBcptSummaryMD\ returned an empty string, the \AnalyzeTests\ step wrote nothing to the job summary, and the build succeeded with no indication that performance tests did not produce output.

Fixes #2158

### Actions/AnalyzeTests/TestResultAnalyzer.ps1

- \GetBcptSummaryMD\: Distinguish between the results file not existing (return empty string, no test was run) and the results file existing but containing no entries (return an informational message). The message is written to the GitHub step summary under the Performance test results heading, making the empty result visible.

### Actions/AnalyzeTests/AnalyzeTests.ps1

- After calling \GetBcptSummaryMD\, check whether the results file is present but the summary is still empty. Emit a \::Warning::\ annotation in that case as a defence-in-depth measure.

### Tests/AnalyzeTests.Test.ps1

- Add tests for \ReadBcptFile\ when the file does not exist and when it exists but contains no entries.
- Add a test for \GetBcptSummaryMD\ verifying that it returns an empty string when the file does not exist and a non-empty warning message when the file exists but contains no entries.

## Testing

1. Create a BCPT test app and configure a suite with codeunit IDs that do not match any published app.
2. Run the AL-Go pipeline so that \RunBCPTTests.ps1\ exits without recording measurements.
3. Before this fix: the Performance test results step writes nothing to the summary and the build succeeds silently.
4. After this fix: the summary shows a message stating that no BCPT results were recorded, and a warning annotation appears on the step.

You can also run the Pester test suite locally:
\\\powershell
Invoke-Pester Tests/AnalyzeTests.Test.ps1
\\\
All 10 tests should pass.

### CODE REVIEW

**Review by aholstrup1 — APPROVED** [review]

File: Tests/AnalyzeTests.Test.ps1 [hunk 1]
```diff
@@ -78,6 +78,35 @@ Describe "AnalyzeTests Action Tests" {
...
+    It 'Test GetBcptSummaryMD returns warning message when file exists but contains no entries' {
+        . (Join-Path $scriptRoot '../AL-Go-Helper.ps1')
+        . (Join-Path $scriptRoot 'TestResultAnalyzer.ps1')
+        $emptyResultsFile = Join-Path ([System.IO.Path]::GetTempPath()) "$([GUID]::NewGuid().ToString()).json"
+        try {
+            $null | ConvertTo-Json -Depth 99 | Set-Content -Path $emptyResultsFile -Encoding UTF8
+            $md = GetBcptSummaryMD -bcptTestResultsFile $emptyResultsFile
+            $md | Should -Not -BeNullOrEmpty
+            $md | Should -Match 'No BCPT results were recorded'
+        }
```
[the remark this answers is not in the document]
Comment (aholstrup1) [reply Tests/AnalyzeTests.Test.ps1:107]: Looks like this was addressed as far as I can tell. Either way I think it's a small enough nitpick that we can resolve it :)

**Review by mazhelez — APPROVED** [review]
